modules/Nncase.Modules.NTT/Passes/AffineSelection/Where.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Canaan Inc. All rights reserved.
+// Licensed under the Apache license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Nncase.IR;
+using Nncase.IR.Affine;
+using Nncase.IR.Tensors;
+
+namespace Nncase.Passes;
+
+public partial class NTTAffineSelectionPass
+{
+    public static bool TryGetWhereAffineRelation(IR.Shape condShape, IR.Shape lhsShape, IR.Shape rhsShape, out AffineDomain[] domains, out AffineMap condMap, out AffineMap lhsMap, out AffineMap rhsMap)
+    {
+        var rank = Math.Max(condShape.Rank, Math.Max(lhsShape.Rank, rhsShape.Rank));
+        domains = IR.F.Affine.Domains(rank);
+        condMap = null!;
+        lhsMap = null!;
+        rhsMap = null!;
```

Comment:
You can use `[MaybeNullWhen(false)]` on `out` parameters to avoid `null!`.

Reply:
resolved